webapp/maps_heatmaps/src/queryDB.ts

```diff
@@ -16,56 +17,57 @@
 
 import { database } from "./index";
 import * as firebase from "firebase";
-import * as geofirestore from "geofirestore";
 import { DateTime } from "./interface";
 
-/* @param center The center of the current map, 
-   @param radius The radius of the circle that contains the current map bounderies
-   @param labels The labels the client queries by
-   @param year, month, day The date the client queries by
+const databaseCollection = database.collection("Images");
+
+/* @param labels The labels the client queries by
+   @param datetime The date the client queries by
+   @param hash The hash of the current map bounderies
    @return The filtered collection by the different queries*/
 function getQueriedCollection(
-  center: firebase.firestore.GeoPoint,
-  radius: number,
   labels: string[],
-  datetime: DateTime
-): geofirestore.GeoQuery {
-  const GeoFirestore = geofirestore.initializeApp(database);
-  const geoCollection = GeoFirestore.collection("images");
-  let dataRef: geofirestore.GeoQuery = geoCollection
-    .near({
-      center: center,
-      radius: radius,
-    })
-    .where("labels", "array-contains-any", labels);
+  datetime: DateTime,
+  hash?: string
+): firebase.firestore.Query {
+  let dataRef = databaseCollection.where(
+    "labels",
+    "array-contains-any",
+    labels
+  );
+  if (hash != undefined) {
+    const hashfield: string = "hashmap.hash" + hash.length;
+    dataRef = dataRef.where(hashfield, "==", hash);
+  }
   if (datetime.year != undefined)
-    dataRef = dataRef.where("year", "==", datetime.year);
+    dataRef = dataRef.where("date.year", "==", datetime.year);
   if (datetime.year != undefined && datetime.month != undefined)
-    dataRef = dataRef.where("month", "==", datetime.month);
+    dataRef = dataRef.where("date.month", "==", datetime.month);
   if (
     datetime.year != undefined &&
     datetime.month != undefined &&
     datetime.day != undefined
   )
-    dataRef = dataRef.where("day", "==", datetime.day);
-  return dataRef;
+    dataRef = dataRef.where("date.day", "==", datetime.day);
+  return dataRef.orderBy("random");
```

Comment:
I think the ordering should be done outside of this function, where it's actually needed.
This would allow to order by something else in another part of the code (e.g. maybe at some point we want to show images ordered  by date in the side panel)

Reply:
removed orderBy("random"),
and moves it to getNextDocs in index.